### PR TITLE
Add Passthrough support

### PR DESCRIPTION
**Pull request by MartinPetkov**
This allows directly passing keys from one recipe to the next.

https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/issues/951

### CODE REVIEW

File: internal/tfengine/tfengine.go [hunk 1]
```diff
@@ -178,6 +179,14 @@ func dumpTemplate(conf *Config, pwd, cacheDir, outputPath string, ti *templateIn
 		return err
 	}
 
+	// Pass through specified keys that should be validated against the
+	// schema.
+	for _, k := range ti.Passthrough {
+		if v, ok := conf.Data[k]; ok {
```
Comment (xingao267) [inline internal/tfengine/tfengine.go:185]: should we throw an error if the specified key in passthrough is invalid?

File: internal/tfengine/schema.go [hunk 2]
```diff
@@ -113,6 +113,21 @@ properties = {
...
         }
+
+        passthrough = {
+          description = <<EOF
+            Keys to pass directly to a child template data.
+            This is not recursive.
+
+            Example:
+              The deployment recipe expects "terraform_addons" to be passed to
+	      it directly, so the project recipe uses this key to do so.
```
Comment (xingao267) [inline internal/tfengine/schema.go:124]: fix indentation

File: internal/tfengine/tfengine.go (hunk 1 above)
Comment (MartinPetkov) [reply internal/tfengine/tfengine.go:185]: Yes, that's probably a good idea. If it's being passthrough'd, then it's probably necessary.

File: internal/tfengine/tfengine.go [hunk 3]
```diff
@@ -178,6 +179,16 @@ func dumpTemplate(conf *Config, pwd, cacheDir, outputPath string, ti *templateIn
...
+			ti.Data[k] = v
+		} else {
+			return fmt.Errorf("did not find key %q to pass through to child template", k)
```
Comment (xingao267) [inline internal/tfengine/tfengine.go:188]: we can probably do 

```
v, ok := conf.Data[k]
if !ok {
   return ...
}
ti.Data[k] = v
```
to reduce indentation

**Review by xingao267 — APPROVED** [review]

**Pull request merged**